### PR TITLE
apps: genericAccessLogParser for nginx_access

### PR DESCRIPTION
**Pull request by braydonk**
The new genericAccessLogParser component works fine for nginx_access and
does not change the output, so switch to using that for
nginx_access component as well.

### CODE REVIEW

File: apps/nginx.go [hunk 1]
```diff
@@ -70,49 +68,7 @@ func (LoggingProcessorNginxAccess) Type() string {
...
-			TimeKey:    "time",
-			TimeFormat: "%d/%b/%Y:%H:%M:%S %z",
-			Types: map[string]string{
-				"http_request_status": "integer",
-				// N.B. "http_request_responseSize" is a string containing an integer.
-				// https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest.FIELDS.response_size
-			},
-		},
-	}.Components(tag, uid)
-	// nginx logs "-" when a field does not have a value. Remove the field entirely when this happens.
```
[the remark this answers is not in the document]
Comment (igorpeshansky) [reply apps/nginx.go:87]: I'd mention both apache and nginx.

File: apps/nginx.go [hunk 2]
```diff
@@ -70,49 +68,7 @@ func (LoggingProcessorNginxAccess) Type() string {
...
-	for _, field := range []string{
-		"host", // always empty with default nginx config
```
Comment (igorpeshansky) [inline apps/nginx.go:89]: This comment is also not present in `genericAccessLogParser`. Not sure how significant it is, just flagging it here. Probably ok to ignore.

File: apps/nginx.go [hunk 3]
```diff
@@ -70,49 +68,7 @@ func (LoggingProcessorNginxAccess) Type() string {
 }
 
 func (p LoggingProcessorNginxAccess) Components(tag string, uid string) []fluentbit.Component {
-	c := confgenerator.LoggingProcessorParseRegex{
-		// Sample line: ::1 - - [26/Aug/2021:16:49:43 +0000] "GET / HTTP/1.1" 200 10701 "-" "curl/7.64.0"
```
Comment (igorpeshansky) [inline apps/nginx.go:74]: Do we want to also add a reference to https://docs.nginx.com/nginx/admin-guide/monitoring/logging/#setting-up-the-access-log in `genericAccessLogParser`?

**Comment by qingling128** [comment]
I'll defer the review of this one to @igorpeshansky

File: apps/nginx.go (hunk 3 above)
Comment (braydonk) [reply apps/nginx.go:74]: Added that alongside the apache docs.

File: apps/nginx.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (braydonk) [reply apps/nginx.go:87]: Mentioned both.

File: apps/nginx.go [hunk 4]
```diff
@@ -70,49 +68,7 @@ func (LoggingProcessorNginxAccess) Type() string {
...
-		// TODO: fluentd's default parser appends (?:\s+(?<http_x_forwarded_for>[^ ]+))? but this is not part of Nginx's log format. Consider adding it or other support for extra fields?
```
[the remark this answers is not in the document]
Comment (igorpeshansky) [reply apps/nginx.go:75]: This seems to be something that both [Apache](https://httpd.apache.org/docs/2.4/mod/mod_proxy.html#x-headers) and [NGINX](https://www.nginx.com/blog/sampling-requests-with-nginx-conditional-logging/) recommend doing for reverse proxy setups. We can wait for someone to request this before adding it to the parser…

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

**Comment by braydonk** [comment]
Thanks @igorpeshansky! Just rebasing on master and re-running the checks.

**Pull request merged**